ext/EnsembleKalmanProcessesMakieExt.jl

```diff
@@ -0,0 +1,688 @@
+(`Makie.Band`)[https://docs.makie.org/dev/reference/plots/band] if
+`plot_std = true`.
+
+Keyword arguments passed to `line_kwargs` and `band_kwargs` are merged with
+`kwargs` when possible. The keyword arguments in `line_kwargs` and
+`band_kwargs` take priority over the keyword arguments in `kwargs`.
+"""
+Visualize.plot_ϕ_mean_over_time(args...; plot_std = false, kwargs...) =
+    _plot_ϕ_mean_over_time(constrainedmeanparamsanditersortime, args...; plot_std, kwargs...)
+
+"""
+    _plot_ϕ_mean_over_time(plot_fn, fig_or_ax, ekp, prior, dim_idx; kwargs...)
+
+Helper function to plot mean constrained parameters against time.
+
+The arguments one can expect is `fig_or_ax`, `ekp`, `prior`, and `dim_idx` or
+the last three arguments.
+"""
+function _plot_ϕ_mean_over_time(plot_fn, args...; plot_std, kwargs...)
+    ekp, prior, dim_idx = last(args, 3)
+    times = accumulate(+, get_Δt(ekp))
+    pushfirst!(times, zero(eltype(times)))
+    plot = plot_fn(args...; plot_std, xvals = times, kwargs...)
+
+    # If an axis is passed in, it must be the first argument. Otherwise, get the
+    # current axis as determined by Makie. Note that current_axis() does not work
+    # if an axis is passed in.
+    ax = typeof(first(args)) <: Makie.AbstractAxis ? first(args) : Makie.current_axis()
+    if plot_std
+        axis_attribs = (
+            xlabel = "Time",
+            title = "Mean and std of ϕ over time [dim $(_get_dim_of_dist(prior, dim_idx))]",
+            ylabel = "$(_get_prior_name(prior, dim_idx))",
+        )
+    else
+        axis_attribs = (
+            xlabel = "Time",
+            title = "Mean ϕ over time [dim $(_get_dim_of_dist(prior, dim_idx))]",
+            ylabel = "$(_get_prior_name(prior, dim_idx))",
+        )
+    end
+    _modify_axis!(ax; axis_attribs...)
+    return plot
+end
+
+"""
+    Makie.plot!(plot::ConstrainedMeanParamsAndItersOrTime)
+
+Plot the mean constrained parameter versus iterations or time.
+"""
+function Makie.plot!(plot::ConstrainedMeanParamsAndItersOrTime)
+    ekp, prior, dim_idx, xvals = plot.ekp[], plot.prior[], plot.dim_idx[], plot.xvals[]
+    ϕ_mean_vals = [get_ϕ_mean(prior, ekp, iter)[dim_idx] for iter in 1:length(xvals)]
+
+    valid_attributes = Makie.shared_attributes(plot, Makie.Lines)
+    hasproperty(plot, :line_kwargs) && (valid_attributes = merge(plot.line_kwargs, valid_attributes))
+    Makie.lines!(plot, xvals, ϕ_mean_vals; valid_attributes...)
+
+    if plot.plot_std[]
+        stds = [std(get_ϕ(prior, ekp, iter)[dim_idx, :]) for iter in 1:length(xvals)]
+
+        valid_attributes = Makie.shared_attributes(plot, Makie.Band)
+        hasproperty(plot, :band_kwargs) && (valid_attributes = merge(plot.band_kwargs, valid_attributes))
+        Makie.band!(plot, xvals, ϕ_mean_vals .- stds, ϕ_mean_vals .+ stds; valid_attributes...)
+    end
+    return plot
+end
+
+"""
+    Visualize.plot_ϕ_over_iters_by_dist(gridpositions, ekp, prior, name; kwargs...)
```

Comment:
Please rename this to `plot_....by_name`  - the "by distribution" doesn't really make sense, as all the parameters come from a distribution.

_Preferably_ (though not required) would be to drop the `by_name` and just accept a string argument (which calls batch(pd) to check the dimension agreement of the grid positions.

If this is too much work here, don't worry - the current implementation is fine along with the docs!